src/autopas/selectors/Smoothing.cpp

```diff
@@ -0,0 +1,157 @@
+/**
+ * @file Smoothing.cpp
+ * @author F. Gratl
+ * @date 23/11/2020
+ */
+
+#include "Smoothing.h"
+
+#include "autopas/utils/Math.h"
+
+/**
+ * Calculates the weights for the k-neighbors of the last point in the vector using the tri-cube function.
+ * If the sum of weights is <= 0 all neighbors are at the same x coordinate as the point and smoothing will not change
+ * the point. This case is indicated by the bool in the return tuple.
+ *
+ * @param points Sorted collection of observations.
+ * @param pointsPerEstimation Number of neighbors to consider for smoothing.
+ * @param maxDistFromIntervalEdge Largest distance between the point that shall be fitted and its neighbors.
+ * @return Tuple of a vector containing the weights for the neighbors
+ * and a bool indicating whether fitting is unnecessary.
+ */
+std::tuple<std::vector<double>, bool> calculateWeightsSimple(const std::vector<std::pair<size_t, size_t>> &points,
+                                                             size_t pointsPerEstimation,
+                                                             size_t maxDistFromIntervalEdge) {
+  const size_t i = points.size() - 1;
+  const size_t firstIndex = i - pointsPerEstimation + 1;
+  // initialize all weights with 0
+  std::vector<double> weights(pointsPerEstimation);
+
+  const auto &xi = points[i].first;
+
+  // define thresholds for shortcuts: if residuals are beyond these values the
+  // are assumed to be 0 respectively 1
+  auto maxDistFromIntervalEdgeHigh = maxDistFromIntervalEdge * .999;
+  auto maxDistFromIntervalEdgeLow = maxDistFromIntervalEdge * .001;
+
+  double sumOfWeights = 0.0;
+
+  // compute all weighs that can be assumed to be non-zero
+  for (size_t j = 0; j < weights.size(); ++j) {
+    auto xj = points[j + firstIndex].first;
+    // residual = |xi - xj|
+    size_t residual = xj > xi ? xj - xi : xi - xj;
```

Comment:
```suggestion
    size_t residual = std::abs(xj - xi);
```

Reply:
this actually does not work since the subtraction of two `size_t` is always positive due to overflow.